### DISCUSSION TITLE
Is there a way to keep software in one process no matter how many times it is opened?

### DISCUSSION THREAD

**Discussion by junmaqiang** [fyne-io/fyne#3671] [Q&A] [answered]
Such as the title

**Comment by Jacalz** [comment, marked as answer]
This is more of a general thing about Go than specific to Fyne from what I can tell. A bit of Googling turned these links up:

- https://stackoverflow.com/questions/23162986/restricting-to-single-instance-of-executable-with-golang
- https://forum.golangbridge.org/t/single-process-instance-achievable-in-go/17564

The short answer is that it is hard to do well in a cross platform way unless you want to write to a file that you check at startup. More information is in those links.

**Reply by junmaqiang** [reply]
thank you ,  I found this https://github.com/marcsauter/single

**Reply by mevdschee** [reply]
I created https://github.com/mevdschee/pidfile for this.
